docs/content/theory/inner_products_index.rst

```diff
@@ -0,0 +1,41 @@
+.. _inner_products_index:
+
```

Comment:
Again here, I would be inclined to remove this page and bring 
- Basic Inner Products 
- Isotropic Constitutive Relationships
- Anisotropic Constitutive Relationships 
- Inner products with differential operators (renaming from "Differential Operators" to distinguish from the "Operators" section)
- Boundary Conditions (which I think should be at top level either way)